app/Main.hs

```diff
@@ -424,38 +423,43 @@ checkForUpdates applyMinorUpdates applyMajorUpdates = do
     isMinorReleaseFrom (x : xs) (y : ys) = y == x && ys > xs
     isMinorReleaseFrom _        _        = False
 
-verify :: String -> IO ()
-verify inputName = case mkPackageName (pack inputName) of
+data VerifyArgs a = Package a | After a | NoArg
+
+verify :: VerifyArgs Text -> IO ()
+verify (Package name) = case mkPackageName name of
   Left pnError -> echoT . pack $ "Error while parsing input package name: " <> show pnError
-  Right name -> do
+  Right pName  -> do
     pkg <- readPackageFile
-    db <- readPackageSet pkg
-    case name `Map.lookup` db of
-      Nothing -> echoT . pack $ "No packages found with the name " <> show (runPackageName $ name)
-      Just _ -> do
-        reverseDeps <- map fst <$> getReverseDeps db name
-        let packages = pure name <> reverseDeps
-        echoT ("Verifying " <> pack (show (length packages)) <> " packages.")
-        echoT "Warning: this could take some time!"
-
-        let installOrUpdate' (name_, pkgInfo) = (name_, ) <$> installOrUpdate (set pkg) name_ pkgInfo
-        paths <- Map.fromList <$> traverse installOrUpdate' (Map.toList db)
-
-        traverse_ (verifyPackage db paths) packages
+    db  <- readPackageSet pkg
+    case pName `Map.lookup` db of
+      Nothing -> echoT . pack $ "No packages found with the name " <> show (runPackageName pName)
+      Just _  -> do
+        reverseDeps <- map fst <$> getReverseDeps db pName
+        let packages = pure pName <> reverseDeps
+        verifyPackages packages db pkg
+
+verify (After after) = case mkPackageName after of
+  Left pnError -> echoT . pack $ "Error while parsing the option argument name: " <> show pnError
+  Right pName  -> do
+    pkg <- readPackageFile
+    db  <- readPackageSet pkg
+    let name     = runPackageName pName
+    let packages = Map.keys $ Map.filterWithKey (\k _ -> runPackageName k >= name) db
+    verifyPackages packages db pkg
 
-verifyPackageSet :: Maybe Text -> IO ()
-verifyPackageSet after = do
+verify NoArg = do
   pkg <- readPackageFile
```

Comment:
These three lines are common to each branch. Maybe the `case` on the `VerifyArgs` argument can be moved inside so that these can be shared code.

Reply:
I spent more time than I care to admit attempting this last night. But that was using three data constructors. I'll try it with two.

Reply:
I'm happy to work on this, by the way. I think this is probably good to merge now, I'm just thinking aloud mostly.

Reply:
No, it was a good lunch activity (and great practice) 😄 

Besides,  TIL `DeriveTraversable `